### PR TITLE
Pieskiuw 48 add support for redis and celery

### PR DESCRIPTION
**Pull request by kamart02**
This PR was a part of https://github.com/Pieski-Uw/Pieski-UW/pull/26, but it was split into two due to its complexity. All of the discussion and changes about redis and celery can be viewed at the provided PR.
@MateuszWasilewski there are some changes in the tests needed. Could you look into it?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,17 @@
...
 
+## Webscraper application
+The webscraper application uses threads not to halt the main django process. Because it uses them to update the database `celery` and `redis` is needed.
+The redis server needs to be set up. Please follow the redis documentation to install the server. If run on ubuntu the redis server can be installed using:  
+```bash
+$ apt install redis 
+```
+The celery process also needs to be started with the Django server:
+```bash
+$ celery -A pieskiUW worker --loglevel=info --concurrency 1 -E
```
Comment (MateuszWasilewski) [inline README.md:15]: I think that this documentation could be improved but it isn't required because docker configuration is planned

File: django/webscraper/models.py [hunk 2]
```diff
@@ -11,63 +10,32 @@ class Pet(models.Model):
...
-    date_out = models.DateField()
-    address_found = models.CharField(max_length=100)
-    box = models.CharField(max_length=50)
-    group_name = models.CharField(max_length=50)
-    group_link = models.URLField()
-    link = models.URLField()
-
-
-class WebscrappingProcess(models.Model):
+    no = models.CharField(max_length=20, null=True)
```
Comment (MateuszWasilewski) [inline django/webscraper/models.py:13]: Is this change required / intentional? It may change DB schema.

**Review by mateuszwasilewski — APPROVED** [review]
This is a big change but is required for proper functionality.
I believe more tasks while scraping could be made async in the future.

**Review by mateuszwasilewski — CHANGES_REQUESTED** [review]
Minor degradation was found while working on docker compose. Will provide fix later today

**Review by mateuszwasilewski — APPROVED** [review]

File: django/webscraper/models.py (hunk 2 above)
Comment (kamart02) [reply django/webscraper/models.py:13]: This is due to the possible errors when scraping animals. During scraping i encountered errors and the easiest fix was to allow null values.

File: django/webscraper/web_scraper.py [hunk 3]
```diff
@@ -92,7 +92,16 @@ def parse_pet(href):
 def __timed_get(href):
     """Performs get request with preset timeout, with a delay"""
     time.sleep(TIMED_GET_DELAY)  # added in order not to get banned from the server
-    return requests.get(href, timeout=TIMED_GET_TIMEOUT)
+    try:
+        return requests.get(
+            href, timeout=TIMED_GET_TIMEOUT, headers={"User-Agent": UserAgent().random}
```
Comment (Sulnek) [inline django/webscraper/web_scraper.py:97]: Changing UserAgent does not impact timeout. Can be deleted
